library/alloc/src/vec.rs

```diff
@@ -1493,6 +1493,16 @@ impl<T> Vec<T> {
     /// static_ref[0] += 1;
     /// assert_eq!(static_ref, &[2, 2, 3]);
     /// ```
+    ///
+    /// As well as `Vec::into_boxed_slice`, any excess capacity will be removed:
+    ///
+    /// ```
+    /// let mut vec = Vec::with_capacity(10);
+    /// vec.extend([1, 2, 3].iter().cloned());
+    /// assert_eq!(vec.capacity(), 10);
+    /// let slice = vec.leak();
+    /// assert_eq!(slice.to_vec().capacity(), 3);
```

Comment:
`to_vec().capacity()` does not really prove anything here since it allocates a new vec and leaves the `&mut [T]` untouched. Thus it doesn't show the size of the underlying allocation.

The resizing is only relevant when one reconstitutes the `Vec` (or a `Box<[]>`) with `from_raw_parts` with capacity == length.

Reply:
@the8472 Thank you for your comment and sorry for the misunderstanding. I will fix the example.